packages/react-query/src/bin/postinstall.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest'
+import { getTanStackReactQueryPackageJson } from './utils/package'
+import { switchVersion } from './utils/switchVersion'
+
+vi.mock('./utils/package')
+vi.mock('./utils/switchVersion')
+
+describe('postinstall', () => {
+  const mockConsoleWarn = vi.spyOn(console, 'warn')
+  const mockGetPackageJson = vi.mocked(getTanStackReactQueryPackageJson)
+  const mockSwitchVersion = vi.mocked(switchVersion)
+
+  const runPostInstall = async (version: string) => {
+    mockGetPackageJson.mockReturnValue({
+      name: 'tanstack-query',
+      version,
+      description: `TanStack Query v${version.split('.')[0]}`,
+    })
+
+    await import('./postinstall')
+  }
+
+  beforeEach(() => {
+    vi.resetModules()
+    vi.clearAllMocks()
+  })
+
+  it('should switch to version 4 when TanStack Query v4 is installed', async () => {
```

Comment:
In my opinion, It's more easy to understand. Could you reflect this review in your pull request please if you okay?

```suggestion
  it('should switch to @suspensive/react-query-4 when @tanstack/react-query@^4 is installed', async () => {
```